lib/api/model/events.dart

```diff
@@ -850,6 +852,68 @@ class UpdateMessageFlagsMessageDetail {
   Map<String, dynamic> toJson() => _$UpdateMessageFlagsMessageDetailToJson(this);
 }
 
+
+/// A Zulip event of type `typing`:
+///   https://zulip.com/api/get-events#typing-start
+///   https://zulip.com/api/get-events#typing-stop
+@JsonSerializable(fieldRename: FieldRename.snake)
+class TypingEvent extends Event {
+  @override
+  @JsonKey(includeToJson: true)
+  String get type => 'typing';
+
+  final TypingOp op;
+  final MessageType messageType;
+  @JsonKey(name: 'sender', readValue: _readSenderId)
+  final int senderId;
```

Comment:
Let's have `_readSenderId` do all of the work of reading the value, instead of sharing it with `name: 'sender'`. That way the reader doesn't have to go back and forth between the two places to see how the work gets done.

Like how in `SubscriptionRemoveEvent`, even though `streamIds` is computed from information at a `subscriptions` key, it doesn't use `name: 'subscriptions'`, and instead the string 'subscriptions' is used in the `_readStreamIds` implementation:

<details>

```dart
/// A [SubscriptionEvent] with op `remove`: https://zulip.com/api/get-events#subscription-remove
@JsonSerializable(fieldRename: FieldRename.snake)
class SubscriptionRemoveEvent extends SubscriptionEvent {
  @override
  @JsonKey(includeToJson: true)
  String get op => 'remove';

  @JsonKey(readValue: _readStreamIds)
  final List<int> streamIds;

  static List<int> _readStreamIds(Map<dynamic, dynamic> json, String key) {
    return (json['subscriptions'] as List<dynamic>)
      .map((e) => (e as Map<String, dynamic>)['stream_id'] as int)
      .toList();
  }

  SubscriptionRemoveEvent({required super.id, required this.streamIds});

  factory SubscriptionRemoveEvent.fromJson(Map<String, dynamic> json) =>
    _$SubscriptionRemoveEventFromJson(json);

  @override
  Map<String, dynamic> toJson() => _$SubscriptionRemoveEventToJson(this);
}
```
</details>

Reply:
I wonder if the `as int` part is necessary for `_readSenderId` because the code generator handles that. Otherwise this refactor should be good.

Reply:
I agree it looks like it's not necessary.